### PR TITLE
added python bulk labeling script

### PR DESCRIPTION
**Pull request by puneetmittal399**

### CODE REVIEW

File: data_labeling_examples/bulk_labeling_python/README.MD [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+CONFIG_PROFILE = "DEFAULT"
+
+# region identifier of DLS Dataset
+REGION_IDENTIFIER = "uk-london-1"
+
+# service_endpoint
+SERVICE_ENDPOINT_DP = f"https://dlstest-dp.{REGION_IDENTIFIER}.oci.oraclecloud.com"
+
+# compartment where DLS Dataset exists
+COMPARTMENT_ID = "ocid1.compartment.oc1..aaaaaaaamrnszmfhl7xzawtef3rl6gmsb62tcoag32yiv5ujorognkuzrqfq"
```
Comment (ananya2812) [inline data_labeling_examples/bulk_labeling_python/README.MD:81]: Why do we need compartment Id as input from user ?

File: data_labeling_examples/bulk_labeling_python/README.MD [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+# ocid of the DLS Dataset
+DATASET_ID = "ocid1.datalabelingdatasetint.oc1.uk-london-1.amaaaaaaniob46iarr2zttq7c5th3jfqwab7d3vrq4daa52tcnnwhkgrowca"
+
+# an array where the elements are all of the labels that you will use to annotate records in your DLS Dataset with. Each element is a separate label.
+LABELS = ["dog", "cat"]
+
+# the algorithm that will be used to assign labels to DLS Dataset records
```
Comment (ananya2812) [inline data_labeling_examples/bulk_labeling_python/README.MD:89]: Please mention possible values for labeling algorithms in this comment.

File: data_labeling_examples/bulk_labeling_python/bulk_labeling_script.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+def label_record(name, id, labeling_algorithm):
+    if labeling_algorithm == "FIRST_REGEX_MATCH":
+        first_match(name=name, record_id=id)
+    elif labeling_algorithm == "FIRST_LETTER_MATCH":
+        first_letter(name=name, record_id=id)
+    elif labeling_algorithm == "CUSTOM_LABELS_MATCH":
+        custom_label_match(name=name, record_id=id)
+
+
+def main():
```
Comment (ananya2812) [inline data_labeling_examples/bulk_labeling_python/bulk_labeling_script.py:134]: Good to add some validation for the inputs provided by the user, like datasetId.

File: data_labeling_examples/bulk_labeling_python/config.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+SERVICE_ENDPOINT_DP = f"https://dlstest-dp.{REGION_IDENTIFIER}.oci.oraclecloud.com"
+# compartment where DLS Dataset exists
+COMPARTMENT_ID = "ocid1.compartment.oc1..aaaaaaaamrnszmfhl7xzawtef3rl6gmsb62tcoag32yiv5ujorognkuzrqfq"
+# ocid of the DLS Dataset
+DATASET_ID = "ocid1.datalabelingdatasetint.oc1.uk-london-1.amaaaaaaniob46iarr2zttq7c5th3jfqwab7d3vrq4daa52tcnnwhkgrowca"
+# an array where the elements are all of the labels that you will use to annotate records in your DLS Dataset with.
+# Each element is a separate label.
+LABELS = ["dog", "cat"]
+# the algorithm that will be used to assign labels to DLS Dataset records
+LABELING_ALGORITHM = "FIRST_LETTER_MATCH"
```
Comment (ananya2812) [inline data_labeling_examples/bulk_labeling_python/config.py:21]: Please mention possible values for labeling algorithms in this comment.

File: data_labeling_examples/bulk_labeling_python/config.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+# config file path
+CONFIG_FILE_PATH = "~/.oci/config"
+# config file profile
+CONFIG_PROFILE = "DEFAULT"
+# region identifier of DLS Dataset
+REGION_IDENTIFIER = "uk-london-1"
+# service_endpoint
+SERVICE_ENDPOINT_DP = f"https://dlstest-dp.{REGION_IDENTIFIER}.oci.oraclecloud.com"
+# compartment where DLS Dataset exists
+COMPARTMENT_ID = "ocid1.compartment.oc1..aaaaaaaamrnszmfhl7xzawtef3rl6gmsb62tcoag32yiv5ujorognkuzrqfq"
```
Comment (ananya2812) [inline data_labeling_examples/bulk_labeling_python/config.py:14]: Why do we need compartment Id as input from user ?

File: data_labeling_examples/bulk_labeling_python/README.MD [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+# maximum number of DLS Dataset records that can be retrieved from the list_records API operation for labeling
+# limit=1000 is the hard limit for list_records
+LIST_RECORDS_LIMIT = 1000
+
+# For CUSTOM_LABEL_MATCH specify the label map
+LABEL_MAP = {"dog/": ["dog", "pup"], "cat/": ["cat", "kitten"]}
+
+# the no of processes to be used for parallel execution by default is set to maximum no of processors in the system
+NO_OF_PROCESSORS = mp.Pool(mp.cpu_count())
```
Comment (chaudharyv) [inline data_labeling_examples/bulk_labeling_python/README.MD:103]: Shouldn't this be in actual code then config?

File: data_labeling_examples/bulk_labeling_python/config.py (hunk 5 above)
Comment (puneetmittal399) [reply data_labeling_examples/bulk_labeling_python/config.py:14]: in list records API we need compartment id and dataset id right?

File: data_labeling_examples/bulk_labeling_python/config.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+LABELING_ALGORITHM = "FIRST_LETTER_MATCH"
+# use for first_match labeling algorithm
+FIRST_MATCH_REGEX_PATTERN = r'^([^/]*)/.*$'
+# maximum number of DLS Dataset records that can be retrieved from the list_records API operation for labeling
+# limit=1000 is the hard limit for list_records
+LIST_RECORDS_LIMIT = 1000
+# For CUSTOM_LABEL_MATCH specify the label map
+LABEL_MAP = {"dog/": ["dog", "tommy"], "cat/": ["cats", "meow"]}
+# the no of processes to be used for parallel execution by default is set to maximum no of processors int he system
+NO_OF_PROCESSORS = mp.Pool(mp.cpu_count())
```
Comment (chaudharyv) [inline data_labeling_examples/bulk_labeling_python/config.py:30]: Same here

File: data_labeling_examples/bulk_labeling_python/README.MD (hunk 2 above)
Comment (puneetmittal399) [reply data_labeling_examples/bulk_labeling_python/README.MD:89]: Comment -> [" ENUMS  for labeling algorithm "FIRST_LETTER_MATCH", "FIRST_REGEX_MATCH",  "CUSTOM_LABELS_MATCH" ]

File: data_labeling_examples/bulk_labeling_python/README.MD (hunk 1 above)
Comment (puneetmittal399) [reply data_labeling_examples/bulk_labeling_python/README.MD:81]: in list records API we need compartment id and dataset id right?

File: data_labeling_examples/bulk_labeling_python/config.py (hunk 4 above)
Comment (puneetmittal399) [reply data_labeling_examples/bulk_labeling_python/config.py:21]: Comment -> [" ENUMS for labeling algorithm "FIRST_LETTER_MATCH", "FIRST_REGEX_MATCH", "CUSTOM_LABELS_MATCH" ]

File: data_labeling_examples/bulk_labeling_python/README.MD (hunk 6 above)
Comment (puneetmittal399) [reply data_labeling_examples/bulk_labeling_python/README.MD:103]: This is the default value I am setting for no_of_processors which will be used in actual code, user can change that if required.

File: data_labeling_examples/bulk_labeling_python/README.MD (hunk 1 above)
Comment (ananya2812) [reply data_labeling_examples/bulk_labeling_python/README.MD:81]: When we validate the datasetId provided by the user, we call get dataset, and in its response we get the compartmentId. Are we not validating user input ?

File: data_labeling_examples/bulk_labeling_python/README.MD (hunk 2 above)
Comment (ananya2812) [reply data_labeling_examples/bulk_labeling_python/README.MD:89]: I meant that you can provide the possible values as comment in the config's comment for easier resference by user.

File: data_labeling_examples/bulk_labeling_python/config.py (hunk 5 above)
Comment (ananya2812) [reply data_labeling_examples/bulk_labeling_python/config.py:14]: When we validate the datasetId provided by the user, we call get dataset, and in its response we get the compartmentId. Are we not validating user input ?

File: data_labeling_examples/bulk_labeling_python/config.py (hunk 4 above)
Comment (ananya2812) [reply data_labeling_examples/bulk_labeling_python/config.py:21]: I meant that you can provide the possible values as comment in the config's comment for easier resference by user.

**Review by ranjeetkgupta — COMMENTED** [review]
1. Check the main method invocation
2. Add comments in the main method invoked.
3. Clean up imports
4. Revisit exception handling.

**Review by ananya2812 — APPROVED** [review]

**Review by chaudharyv — APPROVED** [review]

**Pull request merged**